bucketlistapi/resources.py

```diff
@@ -0,0 +1,323 @@
+from collections import OrderedDict
+from flask import g, request
+from flask_httpauth import HTTPBasicAuth
+from flask_restful import abort, Resource
+from sqlalchemy import and_
+from webargs.flaskparser import use_args
+
+
+from bucketlistapi.utils import (user_reg_login_field, name_field,
+                                 name_done_field, limit_field, save)
+from bucketlistapi.models import BucketList, BucketListItem, User
+from bucketlistapi import db
+
+auth = HTTPBasicAuth()
+
+
+@auth.verify_password
+def verify_password(username_or_token, password):
+    # first try to authenticate by token
+    user = User.verify_auth_token(username_or_token)
+    if not user:
+        # try to authenticate with username/password
+        user = User.query.filter_by(username=username_or_token).first()
+        if not user or not user.verify_password(password):
```

Comment:
This is fine but the following is probably more readable
`if user and user.verify_password(password):
      g.user = user
      return True
  return False
`